### PR TITLE
Attendance function and tests done

### LINKED ISSUE #11: Milestone 2 Deliverables
- [x] Player stats function + tests. #24 
- [x] Team stats function + tests. #32 
- [x] Draft function + tests. #33 
- [ ] Attendance function + tests. #34

### LINKED ISSUE #30: attendance function and its tests for milestone 2
Milestone 2 expected outcomes: competed `attendance` function and 2 unit tests covering at least 2 - 3 edge cases. The requirements are detailed as follows:

- [x] Iteratively code up the function.
- [x] 2 unit tests.
- [x] The unit test should cover at least 2 - 3 edge cases.
- [x] The test(s) should cover all branches of program flow (i.e. both sides of an if-statement).
- [x] Update the documentation in response to the coding and the tests for any changes (function input were changed, remember to update all!).
- [x] Document the test function(s).
- [x] Check the that the code passes the tests.
- [x] Write the test in the pytest module.

Ideas for tests:

- Make sure the end_season is at least equal to or grater than the start_season.
- Ensure the start_season and end_season are of the appropriate type.
- Ensure the `regular` and `playoffs` are boolean values
- Make sure the start_season and end_season are valid dates to query.
  - Update the documentation describing what the valid dates are.
- Make sure the default values for the current season are appropriate (i.e. if the arguments are left blank).
- Check that the function returns plot based on the correct subset of original data frame.
- Check that the API request returns a valid response. 
  - This could either be an assert, and/or a try/catch statement. 
- Implement a system to handle when bad responses are returned (either start_season or end_season is 2004)
- Identify why a bad response was made.

### PR DESCRIPTION
**Pull request by chuusan**
I just added my attendance function and its test. please have a look, thanks

### CODE REVIEW

File: pypuck/pypuck.py [hunk 1]
```diff
@@ -77,22 +78,21 @@ def player_stats(start_date=None, end_date=None):
     return pd.DataFrame(page.json()['data'])
 
 
-def attendance(regular=True, playoffs=True, season=None):
+def attendance(regular=True, playoffs=True, start_season=None, end_season=None):
     """
     Query to the Attendance API to get the NHL's seasonal and playoff attendance numbers.
```
Comment (jnederlo) [inline pypuck/pypuck.py]: @chuusan could you add a little more description to this? For example, query the NHL attendance numbers from __________ to ____________ from the NHL records API. The attendance represents yearly attendance numbers for all teams combined.

The user can specify to return either the regular season attendance numbers, playoff attendance numbers, or both. The function will display a chart showing the attendance over the specified time period.

File: pypuck/pypuck.py [hunk 2]
```diff
@@ -77,22 +78,21 @@ def player_stats(start_date=None, end_date=None):
...
        
-
-
-
     Parameters
     ----------
     regular : boolean (default True).
     
     playoffs : boolean (default True)
```
Comment (jnederlo) [inline pypuck/pypuck.py]: @chuusan I think these should be a little more explicit:

```suggestion
    regular : boolean (default True).
        Whether to query seasonal regular season attendance data.
    playoffs : boolean (default True)
        Whether to query seasonal playoff attendance data.
```

File: pypuck/pypuck.py [hunk 3]
```diff
@@ -101,11 +101,98 @@ def attendance(regular=True, playoffs=True, season=None):
...
+    # Specify the URL
+    url= 'https://records.nhl.com/site/api/attendance'
+
+    # Make the API request
+    page = requests.get(url)
+
+    df = pd.DataFrame(page.json()['data']).sort_values(by=['seasonId'])
+
+    # Check the response code is valid - i.e. the API didn't fail
+    helpers.check_response_code(page.status_code)
```
Comment (jnederlo) [inline pypuck/pypuck.py]: @chuusan Thanks for adding in the helper!! One minor thing. We want the helper to be called before getting the df, so that the function gracefully errors if there is a bad response code:

```suggestion
    # Check the response code is valid - i.e. the API didn't fail
    helpers.check_response_code(page.status_code)
    
    df = pd.DataFrame(page.json()['data']).sort_values(by=['seasonId'])
```

File: pypuck/pypuck.py [hunk 4]
```diff
@@ -101,11 +101,98 @@ def attendance(regular=True, playoffs=True, season=None):
...
+
+    if pd.isnull(end_season):
+        end_season = 2019    
+            
+    # check if a proper input is given
+    if type(regular) != bool:
+      raise Exception('Only boolean value can be accepted')
+
+    if type(playoffs) != bool:
+      raise Exception('Only boolean value can be accepted')
```
Comment (jnederlo) [inline pypuck/pypuck.py]: @chuusan if you wanted I made a helper function for this:

```suggestion
    helpers.check_argument_type(regular, 'regular', bool)
    helpers.check_argument_type(playoffs, 'playoffs', bool)
```

**Review by jnederlo — COMMENTED** [review]
Code looks well scripted and the tests look really good! I have a few suggestions that you can choose to add.

File: pypuck/pypuck.py (hunk 1 above)
Comment (chuusan) [reply pypuck/pypuck.py]: Sure.

**Review by jnederlo — APPROVED** [review]
Tests were constructed well, but had some errors. Some returns were incorrect so tests would have failed when they should have passed. Also, tests didn't work for different version of Altair so I attempted to remedy that. 

I also fixed a small syntax error in the attendance() function which was causing the program to error.
